railties/test/application/test_runner_test.rb

```diff
@@ -229,24 +216,128 @@ def test_env
       assert_match "development", run_test_command('test/unit/env_test.rb')
     end
 
+    def test_run_in_test_environment_by_default
+      create_env_test
+
+      assert_match "Current Environment: test", run_test_command('test/unit/env_test.rb')
+    end
+
     def test_run_different_environment
-      env = "development"
-      app_file 'test/unit/env_test.rb', <<-RUBY
+      create_env_test
+
+      assert_match "Current Environment: development",
+        run_test_command("-e development test/unit/env_test.rb")
+    end
+
+    def test_generated_scaffold_works_with_rails_test
+      create_scaffold
+      assert_match "0 failures, 0 errors, 0 skips", run_test_command('')
+    end
+
+    def test_run_multiple_folders
+      create_test_file :models, 'account'
+      create_test_file :controllers, 'accounts_controller'
+
+      run_test_command('test/models test/controllers').tap do |output|
+        assert_match 'AccountTest', output
+        assert_match 'AccountsControllerTest', output
+        assert_match '2 runs, 2 assertions, 0 failures, 0 errors, 0 skips', output
+      end
+    end
+
+    def test_run_with_ruby_command
+      app_file 'test/models/post_test.rb', <<-RUBY
         require 'test_helper'
 
-        class EnvTest < ActiveSupport::TestCase
-          def test_env
-            puts Rails.env
+        class PostTest < ActiveSupport::TestCase
+          test 'declarative syntax works' do
+            puts 'PostTest'
+            assert true
           end
         end
       RUBY
 
-      assert_match env, run_test_command("-e #{env} test/unit/env_test.rb")
+      Dir.chdir(app_path) do
+        `ruby -Itest test/models/post_test.rb`.tap do |output|
```

Comment:
Also if we ever regress here `RuntimeError` just goes straight through here. I've tried catching it, but I guess the fact that we're in another process(?) throws us off. The tests will be marked as a failure, but we'll never hit the assertions below.